Update setup.py URL and bump version

richford opened the pull request:
This PR updates the URL field in cloudknot/_meta.py and bumps the version.

@arokem could you help me understand the new versioning/publishing procedure? I previously pushed a 0.4.0 tag to master, and the github actions dutifully published to PyPI. Should I also update the version in cloudknot/_meta.py as I'm doing in this PR? And then should I also publish a release on Github as well? It'd be nice not to have to manually update the version number in cloudknot/_meta.py and to publish a release on github with every new tag as well.

It looks like there is a [github action to create github releases](https://github.com/actions/create-release) but I'm not sure how one gets rid of manually incrementing the version number.

richford commented:
And while we're at it, we should consider which badges we want to keep/update in README.md. The first three are pointing to richford resources. If we want to keep them, we should establish the same ones under nrdg. The last one is the zenodo doi badge but it is pointing to the 0.3.1 release. Is there some way to roll the zenodo doi into the new automated release framework?

arokem commented:
Sorry: should have probably provided more information with #227. 

And I messed it up a bit: We should remove the version strings in `_meta.py`. This is because `scm_setuptools` automatically generates a version string (that gets saved in an auto-generated file called `cloudknot/version.py` -- that's where it gets imported from [here](https://github.com/nrdg/cloudknot/blob/master/cloudknot/__init__.py#L17)). The version string gets generated so that it tells you about the SCM state of the repo at the time. If there's a version tag, it will be used as a proper version string. If this is just a merge into master, it will get a `dev` version string. Currently, this is being controlled by [this bit of code](https://github.com/nrdg/cloudknot/blob/master/setup.py#L14-L28), which is a hack. There are better ways to do this, such that the versions on test pypi are stored as serial dev version numbers from a tagged release (akin to what moto has set up). We could follow up to improve this.

arokem commented on cloudknot/_meta.py:
```diff
@@ -4,7 +4,7 @@
 
 # Format expected by setup.py and doc/source/conf.py: string of form "X.Y.Z"
 _version_major = 0
-_version_minor = 3
+_version_minor = 4
 _version_micro = 1  # use '' for first of series, number for 1 and above
 # _version_extra = 'dev'
 _version_extra = ""  # Uncomment this for full releases
```
In other words, please remove these lines.

arokem commented:
To answer this:

> And then should I also publish a release on Github as well? 

Yes, as it is now, after tagging, we still need to create the release on GitHub. Would be nice to automate that.

richford commented:
@arokem I removed the offending lines as well as any others that referenced those variables. The only thing that is left is in setup.py where `version=VERSION`. I was not sure what to put in place of `VERSION`. Should it be `get_version()` or should that line be deleted entirely? Apologies for the friction as I try to grok the new and improved system.

I also added a section for maintainers to CONTRIBUTING.md.

arokem commented:
No - you're good. I simply didn't finish all of the work in #227 

Yes: we need to remove the line that has `version=VERSION` in it. 

Basically, we want the same call as pyAFQ has here: https://github.com/yeatmanlab/pyAFQ/blob/master/setup.py#L58-L78

richford commented:
Added a commit to remove .travis.yml, remove cloudknot/_meta.py from coverage, and actually use the correct version in doc/conf.py.

Lastly, @arokem, we can leave [this import](https://github.com/nrdg/cloudknot/blob/master/cloudknot/__init__.py#L17) in __init__.py as is, correct?

richford commented:
Aha, I used `importlib.metadata` as recommended [here](https://pypi.org/project/setuptools-scm/)

richford commented:
Okay, @arokem, I think this is ready for review. Tests pass on my local machine.

arokem commented on CONTRIBUTING.md:
````diff
@@ -74,3 +74,21 @@ If you are proposing a feature:
+```bash
+git tag <tagname> -a
+```
+which will open up a text editor with which you can type the tag annotation.
````
```suggestion
For releases, use the following format for <tagname>: "v<major>.<minor>.<micro>". 
When executed, this will open up a text editor with which you can type the tag annotation --
typically, something like: "Version <major>.<minor>.<micro> release".
```
